app/views/works/_work_header_navigation.html.erb

```diff
@@ -51,10 +51,15 @@
       <%= link_to ts("Cancel Bookmark"), "#", :class => "hidden bookmark_form_placement_close" %>
     </li>
 
-    <% unless current_user.is_author_of?(@work) || current_user.try(:preference).try(:history_enabled) == false || marked_for_later?(@work) %>
-      <li class="mark"><%= marktoread_link(@work) %></li>
+    <% unless current_user.is_author_of?(@work) || current_user.try(:preference).try(:history_enabled) == false %>
+      <% if  marked_for_later?(@work) %>
```

Comment:
There's an extra space after `if` here. We also don't need to repeat the `<li>` element, since one or the other will always be true:

```
<li class="mark">
  <% if marked_for_later?(@work) %>
    [Mark as Read]
  <% else %>
    [Mark for Later]
  <% end %>
</li>
```

And while I don't mind re-adding a mark as read option here, it should use the existing "Mark as Read" text, not the new "Remove mark for later" text. Also, if we're changing the option up here to show both states, we might want to consider changing the button at the bottom of the work to show them as well.